credential/concept/registry.go

```diff
@@ -0,0 +1,166 @@
+/*
+ * Nuts node
+ * Copyright (C) 2021 Nuts community
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *
+ */
+
+package concept
+
+import (
+	"github.com/nuts-foundation/go-did"
+)
+
+// Registry defines the interface for accessing loaded concepts and using the templates
+// to generate queries and transform results.
+type Registry interface {
+	// ConceptTemplates returns a mapping of concept names to parsed templates.
+	ConceptTemplates() map[string][]Template
+	// Load template files from disk and parse them.
+	LoadTemplates() error
```

Comment:
can't we do this in the `NewRegistry` function? Since it implies a specific implementation detail (concepts must be loaded, probably from disk/URL) and that ideally shouldn't be in a interface describing the contract for a `Registry` imo

Reply:
Placed in in the `NewRegistry` function for now. It'll probably have to move to a sort of configure function in the future. Added a todo for that.